### PR TITLE
Parametrised new tasks flush

### PR DESCRIPTION
**Pull request by marcocast**

### CODE REVIEW

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/core/properties/PASchedulerProperties.java [hunk 1]
```diff
@@ -352,7 +352,9 @@ negative number indicates that session is infinite (value specified in second) *
     SCHEDULER_DB_RECOVERY_LOAD_JOBS_BATCH_SIZE(
             "pa.scheduler.db.recovery.load.jobs.batch_size",
             PropertyType.INTEGER,
-            "100"),
+            "200"),
```
Comment (bamedro) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/core/properties/PASchedulerProperties.java]: Unsure why you update this value in this commit.
I think there is no relation

Comment (marcocast) [reply scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/core/properties/PASchedulerProperties.java]: yes wrong update

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SchedulerDBManager.java#L1737](https://github.com/ow2-proactive/scheduling/blob/6fcc6a7f66480f802daa354cb13219633c9543db/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L1737): Remove this unused private "taskId" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [SchedulerDBManager.java#L1748](https://github.com/ow2-proactive/scheduling/blob/6fcc6a7f66480f802daa354cb13219633c9543db/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L1748): Remove this unused private "jobId" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [SchedulerDBManager.java#L86](https://github.com/ow2-proactive/scheduling/blob/6fcc6a7f66480f802daa354cb13219633c9543db/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L86): Remove this unused import 'org.ow2.proactive.scheduler.core.helpers.JobsMemoryMonitorRunner'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [SchedulerDBManager.java#L87](https://github.com/ow2-proactive/scheduling/blob/6fcc6a7f66480f802daa354cb13219633c9543db/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L87): Remove this unused import 'org.ow2.proactive.scheduler.core.helpers.TableSizeMonitorRunner'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [SchedulerDBManager.java#L106](https://github.com/ow2-proactive/scheduling/blob/6fcc6a7f66480f802daa354cb13219633c9543db/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L106): Remove this unused import 'it.sauronsoftware.cron4j.Scheduler'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SchedulerDBManager.java#L1737](https://github.com/ow2-proactive/scheduling/blob/6fcc6a7f66480f802daa354cb13219633c9543db/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L1737): Remove this unused private "taskId" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [SchedulerDBManager.java#L1748](https://github.com/ow2-proactive/scheduling/blob/6fcc6a7f66480f802daa354cb13219633c9543db/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java#L1748): Remove this unused private "jobId" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 2]
```diff
@@ -1170,7 +1172,7 @@ public Void doInTransaction(Session session) {
                     }
                     TaskData taskData = saveNewTask(session, jobRuntimeData, task);
                     saveSingleTaskDependencies(session, task, taskData);
-                    if (++counter % 50 == 0) {
+                    if (++counter % SCHEDULER_DB_NEW_TASKS_BATCH == 0) {
```
Comment (jrochas) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java]: Let's use `long` type for the counter?

Comment (bamedro) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java]: so we can go beyond 2 billions tasks in a workflow ?

**Pull request closed without merging**